src/generic-provider-views/index.js

```diff
@@ -308,6 +313,51 @@ module.exports = class View {
     return this.plugin.core.getState()[this.plugin.stateId].activeRow === this.plugin.getItemId(file)
   }
 
+  isChecked (file) {
+    return (this.fileToId(file) in this.plugin.core.getState().files)
+  }
+
+  toggleCheckbox (e, file) {
+    e.stopPropagation()
+    e.preventDefault()
+    let {folders, files, filterInput} = this.plugin.core.getState()[this.plugin.stateId]
+    let items = folders.concat(files)
+    if (filterInput !== '') {
+      items = this.filterItems(items)
+    }
+    let itemsToToggle = [file]
+    if (this.lastCheckbox && e.shiftKey) {
+      let prevIndex = items.indexOf(this.lastCheckbox)
+      let currentIndex = items.indexOf(file)
+      if (prevIndex < currentIndex) {
+        itemsToToggle = items.slice(prevIndex, currentIndex + 1)
+      } else {
+        itemsToToggle = items.slice(currentIndex, prevIndex + 1)
+      }
+    }
+    this.lastCheckbox = file
+    if (this.isChecked(file)) {
+      for (let item of itemsToToggle) {
+        const itemId = this.fileToId(item)
+        if (itemId in this.plugin.core.getState().files) {
+          this.plugin.core.removeFile(itemId)
+        }
+      }
+    } else {
+      for (let item of itemsToToggle) {
+        this.addFile(item, true)
+      }
+    }
+  }
+
+  fileToId (file) {
```

Comment:
@sadovnychyi what is the need for the `fileToId` method? Why not just use `this.plugin.getItemId(file)`?

Nice work btw 😉